src/balancer/accept_http.rs

```diff
@@ -309,6 +309,79 @@ macro_rules! fetch_from_rpc {
     }};
 }
 
+macro_rules! fetch_from_rpc_uncached {
+    (
+        $tx:expr,
+        $rpc_list_rwlock:expr,
+        $rpc_position:expr,
+        $route_group:expr,
+        $ttl:expr,
+        $max_retries:expr
+    ) => {{
+        // Loop until we get a response
+        let rx;
+        let mut retries = 0;
+        loop {
+            // Get the next Rpc in line.
+            let mut rpc;
+            {
+                let mut rpc_list = $rpc_list_rwlock.write().unwrap();
+                (rpc, $rpc_position) = pick(&mut rpc_list, &$route_group);
+            }
+            log_info!("Forwarding to: {}", rpc.name);
+
+            // Check if we have any RPCs in the list, if not return error
+            if $rpc_position == None {
+                return (no_rpc_available!(), None);
+            }
+
+            // Send the request. And return a timeout if it takes too long
+            //
+            // Check if it contains any errors or if its `latest` and insert it if it isn't
+            match timeout(
+                Duration::from_millis($ttl.try_into().unwrap()),
+                rpc.send_request($tx.clone()),
+            )
+            .await
+            {
+                Ok(rxa) => {
+                    rx = rxa.unwrap();
+                    break;
+                },
+                Err(_) => {
+                    log_wrn!("\x1b[93mWrn:\x1b[0m An RPC request has timed out, picking new RPC and retrying.");
```

Comment:
Should the string inside be `Warn`?
```suggestion
                    log_wrn!("\x1b[93mWarn:\x1b[0m An RPC request has timed out, picking new RPC and retrying.");
```